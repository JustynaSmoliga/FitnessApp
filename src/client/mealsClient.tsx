import { AxiosResponse } from "axios";
import { Moment } from "moment";

// interface Product {
//   name: string;
//   totalCalories: number;
//   caloriesInGrams: number;
//   quantity: number;
// }

// export interface Meals {
//   date: Date;
//   breakfast: Product[];
//   lunch: Product[];
//   dinner: Product[];
//   supper: Product[];
//   snacks: Product[];
// }

export interface Product {
    name: string;
    totalCalories: number;
    caloriesInGrams: number;
    quantity: number;
  }
  
  export interface Meals {
    date: Moment;
    breakfast: Product[];
    lunch: Product[];
    dinner: Product[];
    supper: Product[];
    snacks: Product[];
  }

const meals = {
  date: new Date(Date.now()),
  breakfast: [
    { name: "egg", totalCalories: 30, caloriesInGrams: 15, quantity: 2 },
    { name: "cucumber", totalCalories: 15, caloriesInGrams: 15, quantity: 1 },
  ],
  lunch: [],
  dinner: [
    { name: "egg", totalCalories: 30, caloriesInGrams: 15, quantity: 2 },
    { name: "cucumber", totalCalories: 15, caloriesInGrams: 15, quantity: 1 },
  ],
  supper: [],
  snacks: [
    {
      name: "popcorn",
      totalCalories: 100,
      caloriesInGrams: 100,
      quantity: 1,
    },
  ],
};

export function getMeals(date:Date): AxiosResponse<Meals>{
//@ts-ignore
    return Promise.resolve({data:meals});
}


