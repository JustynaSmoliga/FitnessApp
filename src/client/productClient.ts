import axios from "axios";

export interface Product {
  id: string;
  name: string;
  totalCalories: number;
  caloriesInGrams: number;
  quantity: number;
}

interface ProductDto {
  id: string;
  name: string;
  kcal: number;
  weightInGrams: number;
}

export async function getProducts(productName: string): Promise<Product[]> {
  const response = await axios.get("http://localhost:8080/products", {
    params: { productName: productName },
  });

  const productDtos: ProductDto[] = response.data;

  const products: Product[] = productDtos.map((productDto) => {
    const product: Product = {
      id: productDto.id,
      caloriesInGrams: productDto.weightInGrams,
      name: productDto.name,
      totalCalories: productDto.kcal,
      quantity: 1,
    };
    return product;
  });
  return products;
}
