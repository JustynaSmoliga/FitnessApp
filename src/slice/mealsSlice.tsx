import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
import moment from "moment";
import {
  addProductToMeal,
  deleteProductFromMeal,
  getMeals,
} from "../client/mealsClient";
import { AddMealProductForm } from "../features/add-meal-product/AddMealProduct";

export interface MealsState extends DayMeals {}

export enum MealType {
  BREAKFAST = "BREAKFAST",
  LUNCH = "LUNCH",
  DINNER = "DINNER",
  SUPPER = "SUPPER",
  SNACKS = "SNACKS",
}

export interface MealProduct {
  id: string;
  name: string;
  kcal: number;
  weightInGrams: number;
}

export interface Meal {
  id: string;
  mealType: MealType;
  mealProducts: MealProduct[];
}

export interface DayMeals {
  date: string;
  breakfast: Meal;
  lunch: Meal;
  dinner: Meal;
  supper: Meal;
  snacks: Meal;
}

const initialState: MealsState = {
  date: moment().toISOString(),
  breakfast: { id: "", mealType: MealType.BREAKFAST, mealProducts: [] },
  lunch: { id: "", mealType: MealType.LUNCH, mealProducts: [] },
  dinner: { id: "", mealType: MealType.DINNER, mealProducts: [] },
  supper: { id: "", mealType: MealType.SUPPER, mealProducts: [] },
  snacks: { id: "", mealType: MealType.SNACKS, mealProducts: [] },
};

export const fetchMeals = createAsyncThunk(
  "meals/fetchMeals",
  async (date: string) => {
    const response = await getMeals(date);
    return response;
  }
);

export const addProduct = createAsyncThunk(
  "meals/addProduct",
  async (product: AddMealProductForm) => {
    const response = await addProductToMeal(product);
    return response;
  }
);

export const deleteProduct = createAsyncThunk(
  "meals/deleteProduct",
  async (productId: string) => {
    const response = await deleteProductFromMeal(productId);
    return response;
  }
);

export const mealsSlice = createSlice({
  name: "meals",
  initialState,
  reducers: {},
  extraReducers: (builder) => {
    builder.addCase(fetchMeals.fulfilled, (state, action) => {
      const meals = action.payload;
      state.date = meals.date;
      state.breakfast = meals.breakfast;
      state.dinner = meals.dinner;
      state.lunch = meals.lunch;
      state.snacks = meals.snacks;
      state.supper = meals.supper;
    });
    builder.addCase(addProduct.fulfilled, (state, action) => {
      const meals = action.payload;
      state.date = meals.date;
      state.breakfast = meals.breakfast;
      state.dinner = meals.dinner;
      state.lunch = meals.lunch;
      state.snacks = meals.snacks;
      state.supper = meals.supper;
    });
    builder.addCase(deleteProduct.fulfilled, (state, action) => {
      const meals = action.payload;
      state.date = meals.date;
      state.breakfast = meals.breakfast;
      state.dinner = meals.dinner;
      state.lunch = meals.lunch;
      state.snacks = meals.snacks;
      state.supper = meals.supper;
    });
  },
});

export default mealsSlice.reducer;
