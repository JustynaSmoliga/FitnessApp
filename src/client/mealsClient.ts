import axios from "axios";
import { DayMeals, Meal } from "../slice/mealsSlice";
import { MealType } from "../slice/mealsSlice";
import apiUrl from "../app/api";
import { AddMealProductForm } from "../features/add-meal-product/AddMealProduct";

interface DayMealsDto {
  id: string;
  date: string;
  meals: { [prop in MealType]: MealDto };
}

interface MealProductDto {
  id: string;
  name: string;
  kcal: number;
  weightInGrams: number;
}

interface MealDto {
  id: string;
  mealType: MealType;
  mealProducts: MealProductDto[];
}

export async function getMeals(date: string): Promise<DayMeals> {
  const response = await axios.get(`${apiUrl}/meals/${date}`, {});

  const dayMealsDto: DayMealsDto = response.data;
  const dayMeals: DayMeals = convertDayMealDtoToDayMeal(dayMealsDto);

  return dayMeals;
}

export async function addProductToMeal(
  product: AddMealProductForm
): Promise<DayMeals> {
  const response = await axios.post(`${apiUrl}/meals/products`, product);
  const dayMealsDto: DayMealsDto = response.data;
  const dayMeals: DayMeals = convertDayMealDtoToDayMeal(dayMealsDto);

  return dayMeals;
}

export async function deleteProductFromMeal(
  productId: string
): Promise<DayMeals> {
  const response = await axios.delete(
    `${apiUrl}/meals/products/${productId}`,
    {}
  );
  const dayMealsDto: DayMealsDto = response.data;
  const dayMeals: DayMeals = convertDayMealDtoToDayMeal(dayMealsDto);
  return dayMeals;
}

function convertDayMealDtoToDayMeal(dayMealDto: DayMealsDto): DayMeals {
  const dayMeals: DayMeals = {
    date: dayMealDto.date,
    breakfast: convertMealDtoToMeal(dayMealDto.meals.BREAKFAST),
    dinner: convertMealDtoToMeal(dayMealDto.meals.DINNER),
    lunch: convertMealDtoToMeal(dayMealDto.meals.LUNCH),
    snacks: convertMealDtoToMeal(dayMealDto.meals.SNACKS),
    supper: convertMealDtoToMeal(dayMealDto.meals.SUPPER),
  };
  return dayMeals;
}

function convertMealDtoToMeal(mealDto: MealDto): Meal {
  const meal: Meal = {
    id: mealDto.id,
    mealType: mealDto.mealType,
    mealProducts: mealDto.mealProducts,
  };
  return meal;
}
