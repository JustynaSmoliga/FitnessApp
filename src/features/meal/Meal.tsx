import { Box, Button, Paper } from "@material-ui/core";
import React, { useState } from "react";
import styles from "./Meal.module.css";
import ProductsListTable from "../products-list-table/ProductsListTable";
import AddMealProduct from "../add-meal-product/AddMealProduct";
import { Meal } from "../../slice/mealsSlice";

export interface MealProps {
  title: string;
  meal: Meal;
}

const MealComponent: React.FC<MealProps> = (props) => {
  const [addMealProductVisible, setAddMealProductVisible] = useState(false);

  const addProductButtonClickHandler = () => {
    setAddMealProductVisible(true);
  };

  return (
    <Box
      display="flex"
      justifyContent="center"
      alignContent="center"
      margin="15px 15px"
    >
      <Paper elevation={3} square>
        <p className={styles.title}>{props.title}</p>
        <Box display="flex" minWidth="550px" justifyContent="center">
          <ProductsListTable productsEaten={props.meal.mealProducts} />
          {addMealProductVisible && (
            <AddMealProduct
              showAddMealProduct={setAddMealProductVisible}
              mealId={props.meal.id}
            />
          )}
        </Box>
        {!addMealProductVisible && (
          <Box paddingBottom="4%" paddingTop="4%">
            <Button
              variant="contained"
              color="secondary"
              type="button"
              onClick={addProductButtonClickHandler}
            >
              Add product
            </Button>
          </Box>
        )}
      </Paper>
    </Box>
  );
};

export default MealComponent;
