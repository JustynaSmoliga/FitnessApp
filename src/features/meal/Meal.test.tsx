import { render, screen, fireEvent, act } from "@testing-library/react";
import moment from "moment";
import { Provider } from "react-redux";
import configureStore, { MockStoreEnhanced } from "redux-mock-store";
import * as MealsSlice from "../../slice/mealsSlice";
import { MealsState, MealType } from "../../slice/mealsSlice";
import Meal, { MealProps } from "./Meal";
import * as ProductClient from "../../client/productClient";

const mockStore = configureStore([]);

const initialState: MealsState = {
  date: moment().toISOString(),
  breakfast: { id: "", mealType: MealType.BREAKFAST, mealProducts: [] },
  lunch: { id: "", mealType: MealType.LUNCH, mealProducts: [] },
  dinner: { id: "", mealType: MealType.DINNER, mealProducts: [] },
  supper: { id: "", mealType: MealType.SUPPER, mealProducts: [] },
  snacks: { id: "", mealType: MealType.SNACKS, mealProducts: [] },
};

describe("meal", () => {
  let store: MockStoreEnhanced<any, any>;
  beforeEach(() => {
    store = mockStore(initialState);
  });

  test('should render "No products in the meal." if no meal products provided', () => {
    const mealProps: MealProps = {
      title: MealType.SUPPER,
      meal: {
        id: "2c6b9dde-cfc0-4971-b6eb-246deb778fce",
        mealType: MealType.SUPPER,
        mealProducts: [],
      },
    };

    render(<Meal title={mealProps.title} meal={mealProps.meal} />);

    expect(screen.getByText("No products in the meal.")).toBeInTheDocument();
  });

  test("should render meal products table", () => {
    const mealProps: MealProps = {
      title: MealType.SUPPER,
      meal: {
        id: "2c6b9dde-cfc0-4971-b6eb-246deb778fce",
        mealType: MealType.SUPPER,
        mealProducts: [
          {
            id: "6fae90ff-7f33-4db6-9770-461a67d371e6",
            name: "carrot",
            kcal: 410,
            weightInGrams: 1000,
          },
          {
            id: "6fae90ff-7f33-4db6-9770-461a67d371e7",
            name: "egg",
            kcal: 100,
            weightInGrams: 1000,
          },
        ],
      },
    };

    const mealComponent = render(
      <Provider store={store}>
        <Meal title={mealProps.title} meal={mealProps.meal} />
      </Provider>
    );

    expect(screen.getByText("carrot")).toBeInTheDocument();
    expect(screen.getByRole("table")).toBeInTheDocument();
    expect(mealComponent).toMatchSnapshot();
  });

  test("should show form after click ADD PRODUCT button", () => {
    const mealProps: MealProps = {
      title: MealType.SUPPER,
      meal: {
        id: "2c6b9dde-cfc0-4971-b6eb-246deb778fce",
        mealType: MealType.SUPPER,
        mealProducts: [
          {
            id: "6fae90ff-7f33-4db6-9770-461a67d371e6",
            name: "carrot",
            kcal: 410,
            weightInGrams: 1000,
          },
          {
            id: "6fae90ff-7f33-4db6-9770-461a67d371e7",
            name: "egg",
            kcal: 100,
            weightInGrams: 1000,
          },
        ],
      },
    };

    render(
      <Provider store={store}>
        <Meal title={mealProps.title} meal={mealProps.meal} />
      </Provider>
    );

    const addProductButton = screen.getByText("Add product");
    addProductButton.click();

    expect(screen.getByRole("form")).toBeInTheDocument();
  });

  test("should dispatch addProduct action after click SAVE button", async () => {
    const mealProps: MealProps = {
      title: MealType.SUPPER,
      meal: {
        id: "2c6b9dde-cfc0-4971-b6eb-246deb778fce",
        mealType: MealType.SUPPER,
        mealProducts: [],
      },
    };

    jest.spyOn(ProductClient, "getProducts").mockResolvedValue([
      {
        id: "id",
        name: "egg",
        kcal: 200,
        weightInGrams: 100,
      },
    ]);
    //@ts-ignore
    jest.spyOn(MealsSlice, "addProduct").mockReturnValue({ type: "TEST" });
    render(
      <Provider store={store}>
        <Meal title={mealProps.title} meal={mealProps.meal} />
      </Provider>
    );

    const addProductButton = screen.getByText("Add product");
    addProductButton.click();
    const searchProduct = screen.getByLabelText("Search product");
    await act(async () => {
      fireEvent.change(searchProduct, { target: { value: "egg" } });
    });
    const selectedValue = screen.getByText("egg");

    selectedValue.click();
    const saveButton = screen.getByText("Save");
    saveButton.click();
    expect(MealsSlice.addProduct).toBeCalledWith({
      mealId: "2c6b9dde-cfc0-4971-b6eb-246deb778fce",
      productId: "id",
      weightInGrams: 100,
    });
    expect(store.getActions().length).toEqual(1);
  });

  test("should dispatch deleteProduct action after click delete button", async () => {
    const mealProps: MealProps = {
      title: MealType.SUPPER,
      meal: {
        id: "2c6b9dde-cfc0-4971-b6eb-246deb778fce",
        mealType: MealType.SUPPER,
        mealProducts: [
          {
            id: "6fae90ff-7f33-4db6-9770-461a67d371e6",
            name: "carrot",
            kcal: 410,
            weightInGrams: 1000,
          },
        ],
      },
    };

    //@ts-ignore
    jest.spyOn(MealsSlice, "deleteProduct").mockReturnValue({ type: "TEST" });

    render(
      <Provider store={store}>
        <Meal title={mealProps.title} meal={mealProps.meal} />
      </Provider>
    );

    const deleteButton = screen.getByRole("button", { name: "deleteButton" });
    deleteButton.click();
    const confirmButton = screen.getByText("AGREE");
    confirmButton.click();

    expect(MealsSlice.deleteProduct).toBeCalledWith(
      "6fae90ff-7f33-4db6-9770-461a67d371e6"
    );
    expect(store.getActions().length).toEqual(1);
  });
});
