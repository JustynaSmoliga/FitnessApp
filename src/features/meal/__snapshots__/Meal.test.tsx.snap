// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`meal should render meal products table 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-4"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation3"
        >
          <p
            class="title"
          >
            SUPPER
          </p>
          <div
            class="MuiBox-root MuiBox-root-5"
          >
            <div
              class="container"
            >
              <table
                class="table"
                id="products"
              >
                <thead>
                  <tr
                    class="header"
                  >
                    <th>
                      Product name
                    </th>
                    <th>
                      Calories[kcal]
                    </th>
                    <th>
                      Weight[g]
                    </th>
                    <th
                      class="deleteButtonHeader"
                    />
                  </tr>
                </thead>
                <tbody>
                  <tr
                    class="row"
                  >
                    <td>
                      carrot
                    </td>
                    <td>
                      410
                    </td>
                    <td
                      class="deleteIconButtonContainer"
                    >
                      1000
                    </td>
                    <td
                      class="deleteButtonDrawer"
                    >
                      <button
                        aria-label="deleteButton"
                        class="MuiButtonBase-root MuiIconButton-root"
                        style="width: 10px; height: 10px;"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            style="font-size: 20px;"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </td>
                  </tr>
                  <tr
                    class="row"
                  >
                    <td>
                      egg
                    </td>
                    <td>
                      100
                    </td>
                    <td
                      class="deleteIconButtonContainer"
                    >
                      1000
                    </td>
                    <td
                      class="deleteButtonDrawer"
                    >
                      <button
                        aria-label="deleteButton"
                        class="MuiButtonBase-root MuiIconButton-root"
                        style="width: 10px; height: 10px;"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            style="font-size: 20px;"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-6"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Add product
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation3"
      >
        <p
          class="title"
        >
          SUPPER
        </p>
        <div
          class="MuiBox-root MuiBox-root-5"
        >
          <div
            class="container"
          >
            <table
              class="table"
              id="products"
            >
              <thead>
                <tr
                  class="header"
                >
                  <th>
                    Product name
                  </th>
                  <th>
                    Calories[kcal]
                  </th>
                  <th>
                    Weight[g]
                  </th>
                  <th
                    class="deleteButtonHeader"
                  />
                </tr>
              </thead>
              <tbody>
                <tr
                  class="row"
                >
                  <td>
                    carrot
                  </td>
                  <td>
                    410
                  </td>
                  <td
                    class="deleteIconButtonContainer"
                  >
                    1000
                  </td>
                  <td
                    class="deleteButtonDrawer"
                  >
                    <button
                      aria-label="deleteButton"
                      class="MuiButtonBase-root MuiIconButton-root"
                      style="width: 10px; height: 10px;"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          style="font-size: 20px;"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </td>
                </tr>
                <tr
                  class="row"
                >
                  <td>
                    egg
                  </td>
                  <td>
                    100
                  </td>
                  <td
                    class="deleteIconButtonContainer"
                  >
                    1000
                  </td>
                  <td
                    class="deleteButtonDrawer"
                  >
                    <button
                      aria-label="deleteButton"
                      class="MuiButtonBase-root MuiIconButton-root"
                      style="width: 10px; height: 10px;"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          style="font-size: 20px;"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-6"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Add product
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
